enhancements/ingress/header-case-adjustment.md

```diff
@@ -0,0 +1,335 @@
+    - Host
+    - Cache-Control
+    - Content-Length
+```
+
+The following Route enables HTTP response header name case adjustments using the
+`haproxy.router.openshift.io/h1-adjust-case` annotation:
+
+```yaml
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  annotations:
+    haproxy.router.openshift.io/h1-adjust-case: true
+  name: my-application
+  namespace: my-application
+spec:
+  to:
+    kind: Service
+    name: my-application
+```
+
+#### Validation
+
+Omitting `spec.httpHeaders` or omitting
+`spec.httpHeaders.headerNameCaseAdjustments` specifies the default behavior.
+
+The API validates that the `spec.httpHeaders.headerNameCaseAdjustments[*]` field
+values are valid HTTP header names as described by the field type's
+`+kubebuilder:validation:Pattern` marker.
+
+The HAProxy configuration template validates the
+`haproxy.router.openshift.io/h1-adjust-case` annotation.  If the annotation
+specifies an invalid value, then the IngressController ignores the annotation.
+
+### User Stories
+
+#### As a cluster administrator, I have a legacy server that expects the HTTP `host` header to be capitalized as `Host`.
+
+To satisfy this use-case, the cluster administrator can specify `Host` in
+`spec.httpHeaders.headerNameCaseAdjustments`:
+
+```
+oc -n openshift-ingress-operator patch ingresscontrollers/default --type=merge --patch='{"spec":{"httpHeaders":{"headerNameCaseAdjustments":["Host"]}}}'
+```
+
+The cluster administrator or application developer can annotate the
+application's Route:
+
+```
+oc annotate routes/my-application haproxy.router.openshift.io/h1-adjust-case=true
+```
+
+The IngressController then adjusts the `host` request header as specified.
+
+#### As a cluster administrator, I have a legacy client that expects the HTTP `cache-control` header to be capitalized as `Cache-Control`.
+
+To satisfy this use-case, the cluster administrator can specify `Cache-Control`
+in `spec.httpHeaders.headerNameCaseAdjustments`:
+
+```
+oc -n openshift-ingress-operator patch ingresscontrollers/default --type=json --patch='[{"op":"add","path":"/spec/httpHeaders/headerNameCaseAdjustments/-","value":"Cache-Control"}]'
+```
+
+The IngressController then adjusts the `cache-control` response header as
+specified.
+
+### Implementation Details
+
+Implementing this enhancement required changes in the following repositories:
+
+* openshift/api
+* openshift/cluster-ingress-operator
+* openshift/router
+
+The router configures HAProxy using a configuration template.  The template uses
+environment variables as input parameters.  The enhancement adds a new
+environment variable, `ROUTER_H1_CASE_ADJUST`, which specifies adjustments as a
+comma-delimited list of header names in the desired capitalizations.  The router
+parses the value of `ROUTER_H1_CASE_ADJUST` and translates it into
+`h1-case-adjust` settings.  If `ROUTER_H1_CASE_ADJUST` is nonempty, the router
+also specifies the `h1-case-adjust-bogus-server` setting.
```

Comment:
```suggestion
also specifies the `h1-case-adjust-bogus-client` setting.
```
The example below shows `h1-case-adjust-bogus-client`, and the example after that shows `h1-case-adjust-bogus-server`, so im assuming you meant `h1-case-adjust-bogus-client` here?

Reply:
Oops!  Well spotted.  I'll fix that.  Thanks!